testsuite/tests/typing-unboxed-types/test_flat.ml

```diff
@@ -193,10 +199,6 @@ type _ s = S : 'a t -> _ s  [@@unboxed]
  and _ t = T : 'a -> 'a s t
 ;;
 [%%expect{|
-Line 1, characters 0-39:
-1 | type _ s = S : 'a t -> _ s  [@@unboxed]
-    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Error: This type cannot be unboxed because
-       it might contain both float and non-float values.
-       You should annotate it with [@@ocaml.boxed].
+type _ s = S : 'a t -> 'b s [@@unboxed]
+and _ t = T : 'a -> 'a s t
```

Comment:
I'd like to see more examples of type definitions that were previously rejected but are now accepted.

For example, this example (from the paper) could be included:

```ocaml
type ('a, 'b) t = K : 'c -> ('c, 'c) t [@@unboxed]
         and t1 = T1 : ('a, int) t -> t1 [@@unboxed]
```

as could @mmottl's example from [Mantis 7364](https://caml.inria.fr/mantis/view.php?id=7364):

```ocaml
type ('a, 'kind) tree =
  | Root : { mutable value : 'a; mutable rank : int } -> ('a, [ `root ]) tree
  | Inner : { mutable parent : 'a node } -> ('a, [ `inner ]) tree

and 'a node = Node : ('a, _) tree -> 'a node [@@ocaml.unboxed]
```

Reply:
I added the two tests that you suggested -- we forgot to do it earlier -- and I changed some other tests to use named instead of perpetually unnamed existential variables. The testsuite still needs some of the more advanced cases that we found while preparing the paper (an example of non-principality, some of the trickier GADT cases, and some cases exercising the support for cyclic/equi-recursive types).

Reply:
I added the tricky tests I had in mind, marking this resolved.